src/metabase/lib/segment.cljc

```diff
@@ -44,24 +44,13 @@
           (lib.metadata.calculation/display-name query stage-number segment-metadata style)))
       (fallback-display-name)))
 
-(mu/defn ^:private filtering-by-segment? :- :boolean
-  "Whether a given stage of a query currently includes a `:segment` ref clause in its filters."
-  [query        :- ::lib.schema/query
-   stage-number :- :int
-   segment-id    :- ::lib.schema.id/segment]
-  (boolean
-   (some (fn [[tag _opts id]]
-           (and (= tag :segment)
-                (= id segment-id)))
-         (:filters (lib.util/query-stage query stage-number)))))
-
 (defmethod lib.metadata.calculation/display-info-method :metadata/segment
-  [query stage-number {:keys [id description], :as segment-metadata}]
+  [query stage-number {:keys [description filter-positions], :as segment-metadata}]
   (let [default-display-info-method (get-method lib.metadata.calculation/display-info-method :default)
         default-display-info        (default-display-info-method query stage-number segment-metadata)]
     (cond-> default-display-info
-      description                                   (assoc :description description)
-      (filtering-by-segment? query stage-number id) (assoc :selected true))))
+      description (assoc :description description)
+      filter-positions (assoc :filter-positions filter-positions))))
```

Comment:
`:filter-positions` is set instead of `:selected`. This is symmetrical to filters.